lib/api/include/irods/get_resource_info_for_operation.h

```diff
@@ -0,0 +1,31 @@
+#ifndef IRODS_GET_RESOURCE_INFO_FOR_OPERATION_H
+#define IRODS_GET_RESOURCE_INFO_FOR_OPERATION_H
+
+struct RcComm;
+struct DataObjInp;
```

Comment:
We need to replace the forward declaration of `DataObjInp` with the appropriate header. Without that, we're forcing devs to manually do it so they can create the `DataObjInp` struct.

```cpp
#include "irods/dataObjInpOut.h"

struct RcComm;
```

Reply:
Hmm, only a forward decl is needed for the `RcComm`, but you need the header for the `DataObjInp`? I don't understand the explanation of why, but it builds the one way or the other, so I'll go ahead and do the requested substitution.

Reply:
The reason for doing that is because developers need to access the members of `DataObjInp` in order to make the API call. The members of the `RcComm` do not need to be known to make the API call.

Including the header for `DataObjInp` keeps the developer from needing to figure out which header defines the data structure. So, it's more about making things convenient.

Reply:
> The reason for doing that is because developers need to access the members of `DataObjInp` in order to make the API call. The members of the `RcComm` do not need to be known to make the API call.
> 
> Including the header for `DataObjInp` keeps the developer from needing to figure out which header defines the data structure. So, it's more about making things convenient.

Makes sense.